File: text-translation-demo/csharp/TextTranslationDemo/TextTranslationClient.cs
using common;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace TextTranslationDemo
{
    class TextTranslationClient
    {
        private string baseUrl;

        private string jwtTokenRequestUrl;

        private int defautDuration = 5 * 60;
        public TextTranslationClient(string baseUrl)
        {
            this.baseUrl = baseUrl;
            this.jwtTokenRequestUrl = this.baseUrl + "/token";
        }
        public async Task<TextTranslationResult[]> TranslateTextBySyncModeAsync(
            ClassiiiUser classiiiUser,
            TextTranslationOption option,
            string[] text)
        {
            string url=baseUrl +"/translate";

            string jwtToken = await HttpUtils.GenerateJwtDataAsync(
                classiiiUser.AccessKey,
                classiiiUser.SecretKey,
                this.defautDuration,
                this.jwtTokenRequestUrl);

            Dictionary<string, object> headers = HttpUtils.BuildJwtHeaders(jwtToken);

            Dictionary<string, object> body = BuildBody(option, classiiiUser.ContractId, text);

            var content = await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new { taskId = "", translationResult = new TextTranslationResult[] { } } };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("text translation failed");

            return serverResp.data.translationResult;
        }

        private static Dictionary<string, object> BuildBody(TextTranslationOption option,string contractId, string[] text)
        {
            Dictionary<string, object> body = new Dictionary<string, object>();
            body.Add("fieldId", option.FieldId.ToString());
            body.Add("targetLang", option.TargetLang);
            body.Add("sourceLang", option.SourceLang);
            body.Add("text", text);
            if (contractId != null)
                body.Add("contractId", contractId);
            if (option.AutoSplit != null)
                body.Add("autoSplit", option.AutoSplit);
            if (option.Type != null)
                body.Add("type", option.Type);
            if (option.RemoveFakeLineBreak != null)
                body.Add("removeFakeLineBreak", option.RemoveFakeLineBreak);
            return body;
        }



        public async Task<string> TranslateTextByAsyncModeAsync(
            ClassiiiUser classiiiUser,
            TextTranslationOption option,
            string[] text)
        {
            string url = baseUrl + "/translate/async";

            string jwtToken = await HttpUtils.GenerateJwtDataAsync(
                classiiiUser.AccessKey,
                classiiiUser.SecretKey,
                this.defautDuration,
                this.jwtTokenRequestUrl);

            Dictionary<string, object> headers = HttpUtils.BuildJwtHeaders(jwtToken);

            Dictionary<string, object> body = BuildBody(option, classiiiUser.ContractId, text);

            var content = await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new { queueId = "" } };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("text translation failed");

            return serverResp.data.queueId;
        }

        public async Task<TextTranslationResult[]> GetTranslationResultAsync(
            ClassiiiUser classiiiUser,            
            string queueId)
        {
            string url = baseUrl + "/translate/async/"+ queueId;

            string jwtToken = await HttpUtils.GenerateJwtDataAsync(
                classiiiUser.AccessKey,
                classiiiUser.SecretKey,
                this.defautDuration,
                this.jwtTokenRequestUrl);

            Dictionary<string, object> headers = HttpUtils.BuildJwtHeaders(jwtToken);

            var content = await HttpUtils.SendAsync(HttpMethod.Get, url, headers, null);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new { taskId = "", translationResult = new TextTranslationResult[] { } } };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("text translation failed");

            if (serverResp.data.taskId == null)
            {
                return new TextTranslationResult[] { };
            }

            return serverResp.data.translationResult;
        }
    }
}

File: text-translation-demo/vb.net/TextTranslationDemo/TextTranslationClient.vb
Imports common.common
Imports Newtonsoft.Json
Imports System.Net.Http
Imports System.Text

Namespace TextTranslationDemo
    Friend Class TextTranslationClient
        Private baseUrl As String

        Public Sub New(ByVal baseUrl As String)
            Me.baseUrl = baseUrl
        End Sub

        Public Async Function TranslateTextBySyncModeAsync(ByVal classiiiUser As ClassiiiUser, ByVal [option] As TextTranslationOption, ByVal text As String()) As Task(Of TextTranslationResult())
            Dim url As String = baseUrl & "/translate"
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(classiiiUser, url)
            Dim body As Dictionary(Of String, Object) = BuildBody([option], classiiiUser.TextContractId, text)
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New With {
                    .taskId = "",
                    .translationResult = New TextTranslationResult() {}
                }
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("text translation failed")
            Return serverResp.data.translationResult
        End Function

        Private Shared Function BuildBody(ByVal [option] As TextTranslationOption, contractId As String, ByVal text As String()) As Dictionary(Of String, Object)
            Dim body As Dictionary(Of String, Object) = New Dictionary(Of String, Object)()
            body.Add("fieldId", [option].FieldId.ToString())
            body.Add("targetLang", [option].TargetLang)
            body.Add("sourceLang", [option].SourceLang)
            body.Add("text", text)
            If Not String.IsNullOrEmpty(contractId) Then body.Add("contractId", contractId)
            If [option].AutoSplit IsNot Nothing Then body.Add("autoSplit", [option].AutoSplit)
            If [option].Type IsNot Nothing Then body.Add("type", [option].Type)
            If [option].RemoveFakeLineBreak IsNot Nothing Then body.Add("removeFakeLineBreak", [option].RemoveFakeLineBreak)
            Return body
        End Function

        Public Async Function TranslateTextByAsyncModeAsync(ByVal classiiiUser As ClassiiiUser, ByVal [option] As TextTranslationOption, ByVal text As String()) As Task(Of String)
            Dim url As String = baseUrl & "/translate/async"
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(classiiiUser, url)
            Dim body As Dictionary(Of String, Object) = BuildBody([option], classiiiUser.TextContractId, text)
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New With {
                    .queueId = ""
                }
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("text translation failed")
            Return serverResp.data.queueId
        End Function

        Public Async Function GetTranslationResultAsync(ByVal classiiiUser As ClassiiiUser, ByVal queueId As String) As Task(Of TextTranslationResult())
            Dim url As String = baseUrl & "/translate/async/" & queueId
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(classiiiUser, url)
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Get, url, headers, Nothing)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New With {
                    .taskId = "",
                    .translationResult = New TextTranslationResult() {}
                }
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("text translation failed")

            If Equals(serverResp.data.taskId, Nothing) Then
                Return New TextTranslationResult() {}
            End If

            Return serverResp.data.translationResult
        End Function
    End Class
End Namespace
